pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -0,0 +1,774 @@
+			if captureState.capturedPacketsNum == captureState.targetCapturedPacketsNum {
+				return nil
+			}
+			captureState.capturedPacketsNum++
+			ci := gopacket.CaptureInfo{
+				Timestamp:     time.Now(),
+				CaptureLength: len(packet.Data()),
+				Length:        len(packet.Data()),
+			}
+			err = captureState.pcapngWriter.WritePacket(ci, packet.Data())
+			if err != nil {
+				return fmt.Errorf("couldn't write packets: %w", err)
+			}
+			klog.V(5).InfoS("Capture packets", "name", captureState.name, "count",
+				captureState.capturedPacketsNum, "len", ci.Length)
+
+			reachTarget := captureState.capturedPacketsNum == captureState.targetCapturedPacketsNum
+			// use rate limiter to reduce the times we need to update status.
+			if reachTarget || captureState.updateRateLimiter.Allow() {
+				pc, err := c.packetCaptureLister.Get(captureState.name)
+				if err != nil {
+					return fmt.Errorf("get PacketCapture failed: %w", err)
+				}
+				// if reach the target. flush the file and upload it.
+				if reachTarget {
+					path := os.Getenv("POD_NAME") + ":" + nameToPath(pc.Name)
+					statusPath := path
+					if err = captureState.pcapngWriter.Flush(); err != nil {
+						return err
+					}
+					if pc.Spec.FileServer != nil {
+						err = c.uploadPackets(pc, captureState.pcapngFile)
+						klog.V(4).InfoS("Upload captured packets", "name", pc.Name, "path", path)
+						statusPath = fmt.Sprintf("%s/%s.pcapng", pc.Spec.FileServer.URL, pc.Name)
+					}
+					// update capture result.
+					if updateErr := c.updatePacketCaptureStatus(pc.Name, captureState.capturedPacketsNum, statusPath, err); updateErr != nil {
+						klog.ErrorS(err, "Failed to update PacketCapture status")
+					}
+					if err != nil {
+						return err
+					}
+					if err := captureState.pcapngFile.Close(); err != nil {
+						klog.ErrorS(err, "Close pcapng file error", "name", pc.Name, "path", path)
+					}
+				}
+
+				err = c.updatePacketCaptureStatus(pc.Name, captureState.capturedPacketsNum, "", nil)
+				if err != nil {
+					klog.ErrorS(err, "Failed to update PacketCapture status")
+				} else {
+					klog.InfoS("Updated PacketCapture", "PacketCapture", klog.KObj(pc), "capturedPacketsNum", captureState.capturedPacketsNum)
+				}
+			}
+		case <-ctx.Done():
+			pc, err := c.packetCaptureLister.Get(captureState.name)
+			if err != nil {
+				return fmt.Errorf("get PacketCapture failed: %w", err)
+			}
+			klog.InfoS("PacketCapture timeout", "name", pc.Name)
+			err = errors.New(captureTimeoutReason)
+			if updateErr := c.updatePacketCaptureStatus(pc.Name, captureState.capturedPacketsNum, "", err); updateErr != nil {
+				klog.ErrorS(updateErr, "The PacketCapture timed out, but PacketCapture status update failed", "name", pc.Name)
+			}
+			return err
+		}
+	}
+}
+
+func (c *Controller) podIP(podRef *crdv1alpha1.PodReference) (net.IP, error) {
+	podInterfaces := c.interfaceStore.GetContainerInterfacesByPod(podRef.Name, podRef.Namespace)
+	var result net.IP
```

Comment:
I suggested to rename `result` as `podIP` but it's not addressed.

Reply:
sorry i thought it was the function name need to change.